dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -57,13 +57,15 @@
   private static final CharMatcher LOWER_ALPHA_NUMERIC =
       CharMatcher.inRange('a', 'z').or(CharMatcher.inRange('0', '9'));
 
+  private final List<UnresolvableArtifactProblem> artifactProblems = new ArrayList<>();
```

Comment:
@elharo This is the reason to make DependencyGraphBuilder non-static. Once DependencyGraphBuilder becomes non-static, then ClassPathBuilder also becomes non-static to hold the instance of DependencyGraphBuilder.

Reply:
Having static `artifactProblem` (a global variable) would cause a problem when we call DependencyGraphBuilder to build a graph multiple times. We cannot distinguish which ArtfiactProblems correspond to which graph building invocation.

Reply:
I agree. That's a reason to not make it a field, static or instance. `artifactProblems` is invocation state, not object state. This can cause problems even in single threaded environments.

Reply:
Let's remove this field.

Reply:
This field has been removed